server/src/main/java/org/eclipse/lsp/cobol/core/visitor/VariableDefinitionDelegate.java

```diff
@@ -115,6 +114,7 @@ public void defineVariable(@NonNull DataDescriptionEntryFormat1Context ctx) {
             .occursClauses(ctx.dataOccursClause())
             .valueClauses(ctx.dataValueClause())
             .usageClauses(ctx.dataUsageClause())
+            .redefinesClauses(ctx.dataRedefinesClause())
```

Comment:
Here you saved `DataRedefinesClauseContext` into `VariableDefinitionContext`, so this will be available later in `defineVariable(` call on the line 135.

On line 135 you can see:
```
    defineVariable(
        variableDefinitionContext,
        this::multiTableDataNameMatcher,
        this::groupItemMatcher,
        this::tableDataNameMatcher,
        this::elementItemMatcher);
```
That means that `variableDefinitionContext` try to be matched on the following variable types. You need to add your own matcher for redefines. Any mathcer starts with `if`. Like this:
```
  private Variable multiTableDataNameMatcher(VariableDefinitionContext variable) {
    if (variable.getPicClauses().isEmpty()
        && !variable.getOccursClauses().isEmpty()
        && variable.getUsageClauses().isEmpty()) {
```
In your case you can check that `variable.getRedefinesClauses() != null`, and please, add it at the top of the matches.

After that, you don't need to have a separate `handleRedefine` method and even store this into the Symantec tree.

Reply:
Thanks @grianbrcom, it is working